### ISSUE TITLE
"error: cannot find value `__f` in this scope" within macro body

### ISSUE THREAD

**Issue by ennis** [mcarton/rust-derivative#54]
**Describe the bug**
The following snippet fails to compile:

```rust
use derivative::Derivative;

macro_rules! gen {
    ($name:ident) => {
        #[derive(derivative::Derivative)]
        #[derivative(Debug)]
        pub struct $name {
            a: i32
        }
    };
}

gen!(Test);

fn main() {
    println!("Hello, world!");
}
```

See error below.

**To Reproduce**
See snippet above.

**Expected behavior**
The snippet should compile and derivative should generate a debug impl for every invocation of `gen`.

**Errors**

```
error[E0425]: cannot find value `__f` in this scope
 --> src\main.rs:5:18
  |
5 |         #[derive(derivative::Derivative)]
  |                  ^ not found in this scope
```

**Version (please complete the following information):**

rustup 1.19.0 (2af131cf9 2019-09-08)
cargo 1.39.0-nightly (3596cb86b 2019-09-19)
rustc 1.39.0-nightly (97e58c0d3 2019-09-20)
derivative 1.0.3

**Additional context**
The error seems to appear only when the struct name is a macro parameter.

**Comment by ennis** [comment]
Same issue on stable (stable-x86_64-pc-windows-msvc - rustc 1.38.0 (625451e37 2019-09-23))

This is a simplified version of a larger piece of code. It used to compile without any issues a few months ago before I updated rust, maybe something changed in rustc in the meantime?

**Comment by mcarton** [comment]
Fixed in https://github.com/mcarton/rust-derivative/pull/60 and released as v2.0.1.
